www/modules/recalls/services/RecallSearchService.js

```diff
@@ -0,0 +1,119 @@
+var main = require('../main'),
+    angular = require('angular');
+
+/**
+ * @class recalls.services.recallSearchService
+ * @see kpmgAngular.services.kHttp
+ * @see https://docs.angularjs.org/guide/services
+ * @see http://docs.angularjs.org/api/ngResource.$resource
+ * @example
+ * function MyCtrl(recallSearchService) {
+ *      recallSearchService.getRecallsByBarcode('111111111111', 0)
+ *          .success(function (result) {
+ *
+ *          })
+ *          .error(function (error) {
+ *
+ *          });
+ * }
+ */
+main.service('recallSearchService', function (/**kpmgAngular.services.kHttp*/ kHttp) {
+
+    var self = this,
+        fdaApiBase = 'https://api.fda.gov/food/enforcement.json',
+        fdaApiKey = 'CGEoOaTA5x5mmrKoA677SU7hW6tLjR94l33eDGic',
+        defaultResultsLimit = 30,
+        defaultRecallStatus = 'ongoing',
+        defaultProductType = 'food';
+
+    /**
+     * Search for recall records based on a UPC code.
+     * @param barcode {string} A UPC barcode string to search for. The string will be matched against code_info.
+     * @param recordToStartWith {number} Of the total number of results, which record should be listed first (e.g. 15 would give you 15 - 45 based on the default result limit).
+     * @returns {{success: Function, error: Function}}
+     */
+    self.getRecallsByBarcode = function (barcode, recordToStartWith) {
+        if(!recordToStartWith) {
+            recordToStartWith = 0;
+        }
+        return kHttp.get(fdaApiBase + '?api_key=:apiKey&limit=:limit&skip=:skip&search=status::status+AND+product_type::type+AND+code_info::barcode', {
```

Comment:
No need to put the :params in the URL field, those will be assumed automatically with the parameters in the lines below. Instead, you should be using :params for the server in place of fdaApiBase.

Example: openfdaConfig.server + '/food/enforcement.json', then in your parameters, you send {api_key: openfdaConfig.apiKey, ...}

Where 'openfdaConfig' is the constant that is exposed in the module.